src/graphics/vulkan.zig

```diff
@@ -203,42 +207,31 @@ pub fn createInstance() void {
 		std.log.debug("\t{s}", .{@as([*:0]const u8, @ptrCast(&ext.extensionName))});
 	}
 
-	// NOTE(blackedout): This array stuff is really ugly but I'm a zig noob :(
-	var extensions = glfwExtensions;
-	var extensionsArray: ?[][*c]const u8 = null;
-	var extensionCount = glfwExtensionCount;
+	// NOTE(blackedout): Add these two extensions to the glfwExtensions and use different create flags if on macOS
+	const extensionsMacOs = [_][*:0]const u8{
```

Comment:
Since these are mac specific, I'd suggest to put them into the macos if branch to make it more readable.